govcd/access_control.go

```
@@ -0,0 +1,230 @@
+			_, seen := used[setting.Subject.HREF]
+			if seen {
+				return fmt.Errorf("[client.SetAccessControl] subject %s (%s) used more than once", setting.Subject.Name, setting.Subject.HREF)
+			}
+			used[setting.Subject.HREF] = true
+			if setting.Subject.Type == "" {
+				return fmt.Errorf("[client.SetAccessControl] subject %s (%s) has no type defined", setting.Subject.Name, setting.Subject.HREF)
+			}
+		}
+	}
+	errorMessage := fmt.Sprintf("[client.SetAccessControl] error setting access control for %s %s (%s)", entityType, entityName, href)
+
+	accessControl.Xmlns = types.XMLNamespaceVCloud
+	resp, err := client.ExecuteRequest(href, http.MethodPost, types.MimeControlAccess,
+		errorMessage+": %s", accessControl, nil)
+
+	if err != nil {
+		return err
+	}
+	if resp == nil {
+		return fmt.Errorf("[client.SetAccessControl] nil response received")
+	}
+	return nil
+}
+
+// GetAccessControl retrieves the access control information for this vApp
+func (vapp VApp) GetAccessControl() (*types.ControlAccessParams, error) {
+
+	if vapp.VApp.HREF == "" {
+		return nil, fmt.Errorf("vApp HREF is empty")
+	}
+
+	return vapp.client.GetAccessControl(vapp.VApp.HREF, "vApp", vapp.VApp.Name)
+}
+
+// SetAccessControl changes the access control information for this vApp
+func (vapp VApp) SetAccessControl(accessControl *types.ControlAccessParams) error {
+
+	if vapp.VApp.HREF == "" {
+		return fmt.Errorf("vApp HREF is empty")
+	}
+
+	return vapp.client.SetAccessControl(accessControl, vapp.VApp.HREF, "vApp", vapp.VApp.Name)
+
+}
+
+// RemoveAccessControl is a shortcut to SetAccessControl with all access disabled
+func (vapp VApp) RemoveAccessControl() error {
+	return vapp.SetAccessControl(&types.ControlAccessParams{IsSharedToEveryone: false})
+}
+
+// IsShared shows whether a vApp is shared or not, regardless of the number of subjects sharing it
+func (vapp VApp) IsShared() bool {
+	settings, err := vapp.GetAccessControl()
+	if err != nil {
+		return false
```

Do we really should hide error here? Maybe as golang allows to return a few fields, return error also.

---

This is a convenient method to get a quick response. You can run `GetAccessControl()` if you want more info.

---

Yep it's convenient, but my question was about if error happens calling `vapp.GetAccessControl()` you getting false as called of IsShared(). Which is not right answer.

---

added error checking